test/test16.c

```diff
@@ -12,6 +12,12 @@ void does_free(char* s CalledMethods("free")) {
     free(s); 
 }
 
+/*
+TODO: change "CalledMethods" to EnsuresCalledMethods / Calls instead of "CalledMethods"
+- with EnsuresCalledMethods assume it's true & check it later 
```

Comment:
"later" isn't quite right: the distinction is between call sites and method definitions. For example, if you have:
```
Calls(x, "free")
void foo(void* x);
```
Then, at each call to `foo(y)`, we assume `y` has had `free()` called on it after `foo()` returns. At the definition of `foo()`, we have to prove that it actually does call `free` on `x`. Which happens first depends on the order in which the analysis visits the files (and the order is random, so we cannot rely on it either way).